snuba/migrate.py

```diff
@@ -14,11 +34,84 @@ def _run_schema(conn, schema):
         return
     clickhouse_table = schema.get_local_table_name()
 
+    STRIP_CAST_RE = re.compile(r"^CAST\((.*), (.*)\)$", re.IGNORECASE)
+    FIXED_STRING_RE = re.compile(r"^FixedString\((\d+)\)$")
+    ENUM_RE = re.compile(r"^Enum\d+\((.*)\)$")
+    ENUM_VALUES_RE = re.compile(r"'(.+)' = (\d+)$")
+    MODIFIER_COL_RE = re.compile(r"^(Array|LowCardinality|Nullable)\((.+)\)$")
+
+    def strip_cast(default_expr: str) -> str:
```

Comment:
Have you consider writing a grammar for this? It seems it would be easier to extend and manage than the ad hoc logic.
Like we did here  https://github.com/getsentry/snuba/blob/master/snuba/query/parser/expressions.py#L18-L43
It would save you all the manual regex matching, and all the custom logic to decide decide when to look for a modifier, which is very hard to verify.

Reply:
I will give it a shot to see how it compares